File: CompetitionSpring/src/main/java/soprajc/CompetitionSpring/model/Reservation.java
package soprajc.CompetitionSpring.model;

import java.time.LocalDate;
import java.time.LocalTime;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.EnumType;
import javax.persistence.Enumerated;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.Version;
import javax.validation.constraints.NotEmpty;
import javax.validation.constraints.NotNull;

import com.fasterxml.jackson.annotation.JsonView;

@Entity
public class Reservation {
	
	@JsonView({JsonViews.Common.class, JsonViews.CompteWithReservation.class})
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Integer id;
	
	@JsonView({JsonViews.Common.class, JsonViews.CompteWithReservation.class})
	@Enumerated(EnumType.STRING)
	@Column(columnDefinition = "ENUM('En_Cours','A_Venir','Termine')")
	private Statut statut;
	
	@JsonView({JsonViews.Common.class, JsonViews.CompteWithReservation.class})
	//@NotNull
	private LocalDate date;
	
	@JsonView({JsonViews.Common.class, JsonViews.CompteWithReservation.class})
	//@NotNull
	private LocalTime heure;
	
	//@JsonView(JsonViews.Common.class)
	@ManyToOne
	@JoinColumn(name="compte_fk")
	private Compte compte;
	
	@JsonView(JsonViews.ReservationWithEpreuve.class)
	@ManyToOne
	@JoinColumn(name="epreuve_fk")
	private Epreuve epreuve;
	
	@JsonView({JsonViews.Common.class, JsonViews.CompteWithReservation.class})
	@ManyToOne
	@JoinColumn(name="logement_fk")
	private Logement logement;
	
	@JsonView({JsonViews.Common.class, JsonViews.CompteWithReservation.class})
	private LocalDate dateDebut;
	
	@JsonView({JsonViews.Common.class, JsonViews.CompteWithReservation.class})
	private LocalDate dateFin;
	
	@Version
	protected int version;
	
	public Reservation() {
	}
	
	public Reservation(Compte compte, Logement logement) {
		this.compte = compte;
		this.logement = logement;
		this.statut=Statut.A_Venir;
		this.date=LocalDate.now();
		this.heure=LocalTime.now();
	}
	

	public Reservation(Compte compte, Epreuve epreuve) {
		this.compte = compte;
		this.epreuve = epreuve;
		this.statut=Statut.A_Venir;
		this.date=LocalDate.now();
		this.heure=LocalTime.now();
	}


	public Integer getId() {
		return id;
	}


	public Statut getStatut() {
		return statut;
	}


	public LocalDate getDate() {
		return date;
	}


	public LocalTime getHeure() {
		return heure;
	}


	public Compte getCompte() {
		return compte;
	}


	public Epreuve getEpreuve() {
		return epreuve;
	}


	public Logement getLogement() {
		return logement;
	}


	public void setId(Integer id) {
		this.id = id;
	}


	public void setStatut(Statut statut) {
		this.statut = statut;
	}


	public void setDate(LocalDate date) {
		this.date = date;
	}


	public void setHeure(LocalTime heure) {
		this.heure = heure;
	}


	public void setCompte(Compte compte) {
		this.compte = compte;
	}


	public void setEpreuve(Epreuve epreuve) {
		this.epreuve = epreuve;
	}


	public void setLogement(Logement logement) {
		this.logement = logement;
	}
	
	public int getVersion() {
		return version;
	}

	public void setVersion(int version) {
		this.version = version;
	}


	public LocalDate getDateDebut() {
		return dateDebut;
	}

	public void setDateDebut(LocalDate dateDebut) {
		this.dateDebut = dateDebut;
	}

	public LocalDate getDateFin() {
		return dateFin;
	}

	public void setDateFin(LocalDate dateFin) {
		this.dateFin = dateFin;
	}

	@Override
	public String toString() {
		return "Reservation [id=" + id + ", statut=" + statut + ", date=" + date + ", heure=" + heure + ", compte="
				+ compte + ", epreuve=" + epreuve + ", logement=" + logement + "]";
	}
	
	
	
	

}

File: CompetitionAngular/src/app/model/reservation.ts
import { Compte } from './compte';
import { Logement } from 'src/app/model/logement';
import { Epreuve } from './epreuve';
import { StatutEvent } from './statut-event';
import { Time } from '@angular/common';

export class Reservation {
  public constructor(
    private _id?: number | undefined,
    private _date?: Date | undefined,
    private _dateDebut?: Date | undefined,
    private _dateFin?: Date | undefined,
    private _heure?: Time | undefined,
    private _statut?: StatutEvent | undefined,
    private _epreuve?: Epreuve | undefined,
    private _logement?: Logement | undefined,
    private _compte?: Compte | undefined
  ) {}

  public get id(): number | undefined {
    return this._id;
  }

  public set id(value: number | undefined) {
    this._id = value;
  }

  public get date(): Date | undefined {
    return this._date;
  }

  public set date(value: Date | undefined) {
    this._date = value;
  }

  public get dateDebut(): Date | undefined {
    return this._dateDebut;
  }

  public set dateDebut(value: Date | undefined) {
    this._dateDebut = value;
  }

  public get dateFin(): Date | undefined {
    return this._dateFin;
  }

  public set dateFin(value: Date | undefined) {
    this._dateFin = value;
  }

  public get heure(): Time | undefined {
    return this._heure;
  }

  public set heure(value: Time | undefined) {
    this._heure = value;
  }

  public get statut(): StatutEvent | undefined {
    return this._statut;
  }

  public set statut(value: StatutEvent | undefined) {
    this._statut = value;
  }

  public get epreuve(): Epreuve | undefined {
    return this._epreuve;
  }

  public set epreuve(value: Epreuve | undefined) {
    this._epreuve = value;
  }

  public get logement(): Logement | undefined {
    return this._logement;
  }

  public set logement(value: Logement | undefined) {
    this._logement = value;
  }

  public get compte(): Compte | undefined {
    return this._compte;
  }

  public set compte(value: Compte | undefined) {
    this._compte = value;
  }
}
